apps/core/components/rangeDatePicker/RenderDay.js

```diff
@@ -0,0 +1,96 @@
+// @flow
+
+import * as React from 'react';
+import { View } from 'react-native';
+import { StyleSheet } from '@kiwicom/universal-components';
+import { TouchableWithoutFeedback, Text } from '@kiwicom/margarita-components';
+import { defaultTokens } from '@kiwicom/orbit-design-tokens';
+import { format, isSameDay, isBefore } from 'date-fns';
+import { margaritaTokens } from '@kiwicom/margarita-design-tokens';
+
+import RangeDateConfig from './RangeDateConfig';
+
+type Props = {|
+  +day: Date,
+  +onPress: Date => void,
+  +price?: ?string,
+  +selectedDate: Date,
+|};
+
+const DayPrice = ({ price }) => (
+  <View>
+    <Text numberOfLines={1} style={styles.price}>
```

Comment:
Performance issue seems to be related to this component. Basically delay increases with the number of rendered days. When I've quickly tried to remove extra wrapper and used `Text` component directly from `react-native` picker seems to be slightly quicker.

Reply:
I don't see any real difference while using `Text` from react native.

When I render only one month of calendar, it's quicker (but still not perfect). 

It's also probably not related with  generating of calendar, because when I use dummy data, delay is practically identical like with real data. So there could be some problem with too many number of elements, Flatlist or mapping weeks into the FlatList. It's probably not related to overdraw.

Reply:
Right, it seems to be more likely related to number of elements.
Text itself is a small fraction of it. When I set number of months to some high value like 50. Show delay decreases with every element removed from `RenderDay` component. And when I keep just something like `return <Text>12</Text>` inside it is relatively quick even with all those 1500 days.
FlatList can be definitely worth a try. It seems to work perfectly for ResultsList where we are rendering probably even more components.

Reply:
So I believe it should be fine now.

There are following changes:
- calculations of day matrixes were moved from `RangeDatePickerContent` to `RenderMonth` (which is inside on `FlatList`) and `FlatList` got following prop: `initialNumToRender={2}` (thanks @tbergq for the advice)
- implemented own `shouldComponentUpdate` in `RenderMonth`
- use `memoize` for rendering in `RenderMonth`
- fix overdraw in the calendar

Calendar is also faster on **production mode**

Reply:
Thanks, it definitely feels quicker now. And you are right that production mode will also help.